code/week3/week3.cpp

```diff
@@ -0,0 +1,274 @@
+     using namespace std;
+
+     // ******************** ARRAYS *********************
+     // array docs http://en.cppreference.com/w/cpp/container/array
+
+     // initialization
+     // array<TYPE, SIZE> name = {elements}
+     array<int, 3> int_array = {1, 2, 3};
+     array<string, 3> string_array = {"arrays", "are", "cool"};
+
+     // You can get the number fo elemnts in std::array by using .size()
+     cout << "size of the array is = " << int_array.size() << endl;
+
+     // remember arrays index 0,1,2,...
+     cout << "printing int_array" << endl;
+     for(int i = 0; i < int_array.size(); i++) {
+         // .at(index) returns the element at that index with bounds checking at runtime
+         cout << "int_array at index = " << i << " the value is = " << string_array.at(i) << endl;
+         cout << "string_array at index = " << i << " the value is = " << string_array.at(i) << endl;
+
+         // [] operator gets the value at the index without bounds checking
+         cout << "int_array at index = " << i << " the value is = " << string_array[i] << endl;
+         cout << "string_array at index = " << i << " the value is = " << string_array.at(i) << endl;
+     }
+     cout << "\n\n";
+
+     // convenience methods exist for getting the front and back elements
+     cout << "front = " << int_array.front() << endl;
+     cout << "back = " << int_array.back() << endl;
+
+     // .empty returns true if the array is empty and false otherwise
+     if(int_array.empty()) {
+         cout << "int_array is not empty" << endl;
+     } else {
+         cout << "int_array is empty" << endl;
+     }
+
+     // ******************** VECTORS *********************
+     // vector docs http://en.cppreference.com/w/cpp/container/vector
+     cout << "\n\n";
+
+     //initialization
+     // vector<TYPE> name = {values ...}
+     vector<char> robo_vector;
+
+     // You can add elements to a vector
+     // push back appends the value to the back of the list
+     robo_vector.push_back('o');
+
+     // size returns the size
+     cout << "size of robo_vector = " << robo_vector.size() << endl;
+
+     // empty determines if the size is null
+     if(!robo_vector.empty()) {
+         robo_vector.push_back('j');
+     }
+
+     // elements can be accessed using at with bounds checking
+     cout << "robo_vector at index 1 is = " << robo_vector.at(1) << endl;
+
+     // elements can be accessed using [] without bounds checking
+     cout << "robo_vector at index 0 is = " << robo_vector[0] << endl;
+
+     // front and back also exist
+     if(robo_vector.size() > 1 && robo_vector.front() != robo_vector.back()) {
+         robo_vector.push_back('a');
+     }
+
+     // the last element can be removed using pop_back
+     robo_vector.pop_back();
+     cout << "size is now = " << robo_vector.size() << endl;
+
+     // resize can be used to change the size of the vector
+     // increasing the size appends the value givent
+     robo_vector.resize(11, 'a');
```

Comment:
It might be useful to print the arrays and vectors to see how the transformations affect them. Maybe make a helper function (that you can treat like a black box)?